src/gui/layertree/qgslayertreeviewdefaultactions.h

```diff
@@ -66,7 +66,11 @@ class GUI_EXPORT QgsLayerTreeViewDefaultActions : public QObject
     QAction *actionZoomToSelection( QgsMapCanvas *canvas, QObject *parent = nullptr ) SIP_FACTORY;
     QAction *actionZoomToGroup( QgsMapCanvas *canvas, QObject *parent = nullptr ) SIP_FACTORY;
 
-    QAction *actionMakeTopLevel( QObject *parent = nullptr ) SIP_FACTORY;
+    /**
+     * \see ungroup()
+     * \since QGIS 3.2
+     */
+    QAction *actionUngroup( QObject *parent = nullptr ) SIP_FACTORY;
```

Comment:
We can't change these names in the API, because they are part of the stable API now.

There's also the risk that someone has code which relies on the old behaviour of the actionMakeTopLevel call.

What I'd suggest here is adding ungroup as a new method/action, and the marking makeTopLevel as deprecated.

The old action could exist, but not actually be inserted into the menu.

Then existing code will work without issue and we can drop the deprecated methods at 4.0

Reply:
Sounds like a good idea. I will make the changes.